src/components/bmfont-text.js

```diff
@@ -0,0 +1,333 @@
+    } else {
+      this.material.uniforms.opacity.value = this.data.opacity;
+      this.material.uniforms.color.value.set(this.data.color);
+      this.material.uniforms.map.value = this.texture;
+    }
+
+    if (this.mesh) { this.mesh.material = this.material; }
+  },
+
+  registerFont: function (key, url) { fontMap[key] = url; },
+
+  lookupFont: function (keyOrUrl) { return fontMap[keyOrUrl] || keyOrUrl; },
+
+  loadBMFontPromise: function (src) {
+    return new Promise(function (resolve, reject) {
+      loadBMFont(src, function (err, font) {
+        if (err) { reject(err); } else { resolve(font); }
+      });
+    });
+  },
+
+  loadTexturePromise: function (src) {
+    return new Promise(function (resolve, reject) {
+      var loader = new THREE.ImageLoader();
+      loader.load(src, function (image) {
+        resolve(image);
+      }, undefined, function () {
+        reject(null);
+      });
+    });
+  },
+
+  updateFont: function () {
+    if (!this.data.font) {
+      console.error(new TypeError('No font specified for bmfont text!'));
+      return;
+    }
+
+    var geometry = this.geometry;
+    var self = this;
+    this.mesh.visible = false;
+    var font = this.lookupFont(this.data.font);
+    var promise = loadedFontPromises[font] = loadedFontPromises[font] || this.loadBMFontPromise(font);
+    promise.then(function (font) {
+      if (font.pages.length !== 1) {
+        throw new Error('Currently only single-page bitmap fonts are supported.');
+      }
+      var data = self.coerceData(self.data);
+
+      var src = self.data.fontImage || self.lookupFont(self.data.font).replace('.fnt', '.png') || path.dirname(data.font) + '/' + font.pages[0];
+      var textRenderWidth = data.wrappixels || (data.wrapcount * 0.6035 * font.info.size);
+      var options = assign({}, data, { font: font, width: textRenderWidth, lineHeight: data.lineHeight || font.common.lineHeight });
+      var object3D;
+      geometry.update(options);
+      self.mesh.geometry = geometry;
+
+      object3D = self.el.object3D;
+      if (object3D.children.indexOf(self.mesh) === -1) {
+        object3D.add(self.mesh);
+      }
+
+      var texpromise = loadedTexturePromises[src] || self.loadTexturePromise(src);
+      texpromise.then(function (image) {
+        self.mesh.visible = true;
+        if (image) {
+          self.texture.image = image;
+          self.texture.needsUpdate = true;
+        }
+      }).catch(function () {
+        console.error('Could not load bmfont texture "' + src +
+          '"\nMake sure it is correctly defined in the bitmap .fnt file.');
+      });
+
+      self.currentFont = font;
+      self.updateLayout(data);
+    }).catch(function (error) {
+      throw new Error('Error loading font ' + self.data.font +
+          '\nMake sure the path is correct and that it points' +
+          ' to a valid BMFont file (xml, json, fnt).\n' + error.message);
+    });
+  },
+
+  updateLayout: function (data) {
+    var el = this.el;
```

Comment:
Yeah maybe not for this case for core http://caniuse.com/#search=const

I usually use it in components that are VR-specific though since we have the benefit of targeting experimental browsers